### PR TITLE
DC-2850 - Expose CDS get email endpoint via channel preferences

### PR DESCRIPTION
**Pull request by jim-collins**

### CODE REVIEW

File: app/uk/gov/hmrc/channelpreferences/hub/cds/services/CdsPreference.scala [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+class CdsPreferenceService @Inject()(cdsEmailCon: CDSEmailConnector) extends CdsPreference {
+  private val log: LoggerLike = Logger(this.getClass)
+  private val HMRC_CUS_ORG = "HMRC-CUS-ORG"
+  private val EORINumber = "EORINumber"
+  def getPreference(c: Channel, enrolmentKey: String, taxIdName: String, taxIdValue: String)(
+    implicit hc: HeaderCarrier): Future[Either[Int, EmailVerification]] =
+    c match {
+      case Email => emailPreference(enrolmentKey, taxIdName, taxIdValue)
+      case _ =>
+        log.info(s"channel $c not implemented")
```
Comment (elabeca) [inline app/uk/gov/hmrc/channelpreferences/hub/cds/services/CdsPreference.scala:46]: should this be a log.error?

File: app/uk/gov/hmrc/channelpreferences/hub/cds/services/CdsPreference.scala [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        Future.successful(Left(NOT_IMPLEMENTED))
+    }
+
+  private def emailPreference(enrolmentKey: String, taxIdName: String, taxIdValue: String)(
+    implicit hc: HeaderCarrier): Future[Either[Int, EmailVerification]] =
+    (enrolmentKey, taxIdName) match {
+      case (HMRC_CUS_ORG, EORINumber) => cdsEmailCon.getVerifiedEmail(taxIdValue)
+      case _ =>
+        log.info(s"($enrolmentKey,$taxIdName) not supported")
```
Comment (elabeca) [inline app/uk/gov/hmrc/channelpreferences/hub/cds/services/CdsPreference.scala:55]: log.error or log.warn?

File: app/uk/gov/hmrc/channelpreferences/hub/cds/services/CdsPreference.scala (hunk 1 above)
Comment (jim-collins) [reply app/uk/gov/hmrc/channelpreferences/hub/cds/services/CdsPreference.scala:46]: I will change it to error

File: app/uk/gov/hmrc/channelpreferences/hub/cds/services/CdsPreference.scala (hunk 2 above)
Comment (jim-collins) [reply app/uk/gov/hmrc/channelpreferences/hub/cds/services/CdsPreference.scala:55]: I will change to error

**Review by elabeca — APPROVED** [review]

File: app/uk/gov/hmrc/channelpreferences/config/AppConfig.scala [hunk 3]
```diff
@@ -27,4 +27,5 @@ class AppConfig @Inject()(config: Configuration, servicesConfig: ServicesConfig)
 
   val auditingEnabled: Boolean = config.get[Boolean]("auditing.enabled")
   val graphiteHost: String = config.get[String]("microservice.metrics.graphite.host")
+  val cdsBaseUrl: String = config.get[String]("microservice.metrics.graphite.host")
```
Comment (lxol) [inline app/uk/gov/hmrc/channelpreferences/config/AppConfig.scala:30]: cds has nothing to do with graphite.
it should be microservice.services.customs-data-store
You should use baseUrl method from service-config to get url not config.get
[https://github.com/hmrc/bootstrap-play/blob/7bae9ce33c27fbbf2677d2e39312698008b6885b/bootstrap-common-play-26/src/main/scala/uk/gov/hmrc/play/bootstrap/config/ServicesConfig.scala#L38](baseUrl)

**Review by lxol — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/channelpreferences/config/AppConfig.scala (hunk 3 above)
Comment (jim-collins) [reply app/uk/gov/hmrc/channelpreferences/config/AppConfig.scala:30]: I have removed this line it is not used. I am using the baseURL method.

**Review by lxol — APPROVED** [review]

**Pull request merged**